src/BlazorWebView/src/SharedSource/BlazorWebViewInitializedEventArgs.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Threading.Tasks;
+#if WEBVIEW2_WINFORMS
+using Microsoft.Web.WebView2.Core;
+using WebView2Control = Microsoft.Web.WebView2.WinForms.WebView2;
+#elif WEBVIEW2_WPF
+using Microsoft.Web.WebView2.Core;
+using WebView2Control = Microsoft.Web.WebView2.Wpf.WebView2;
+#elif WINDOWS && WEBVIEW2_MAUI
+using Microsoft.Web.WebView2.Core;
+using WebView2Control = Microsoft.UI.Xaml.Controls.WebView2;
+#elif ANDROID
+using AWebView = Android.Webkit.WebView;
+#elif IOS || MACCATALYST
+using WebKit;
+#endif
+
+namespace Microsoft.AspNetCore.Components.WebView
+{
+	/// <summary>
+	/// Allows configuring the underlying web view after it has been initialized.
+	/// </summary>
+	public partial class BlazorWebViewInitializedEventArgs : EventArgs
+	{
+#nullable disable
+#if WINDOWS
+		/// <summary>
+		/// Gets or sets a function that will be invoked once the web view has been initialized with
+		/// the default values to allow further configuring additional options.
+		/// </summary>
+		public WebView2Control WebView { get; internal set; }
```

Comment:
Not a fan that the setter is internal. That means no one else can create a meaningful instance of this args type. While surely not common, it would mean you couldn't easily write a unit test for your event handler because you can't construct this event args.

Could we add ctor's for each of these? (Yes, I know it's not pretty to have 1 new constructor for each, but I think it's the right thing to do.)

Reply:
Could we add the constructor once we need it?

I'm skeptical of people writing unit tests involving this component and I would like to see a concrete scenario first.

Reply:
I think this is an anti-pattern but I don't have any other strong argument for it. It's very odd to have a type no one can construct. I won't block on this.